components/wallet-logger.js

```diff
@@ -313,9 +315,16 @@ export function useWalletLogs (wallet, initialPage = 1, logsPerPage = 10) {
     }
   }, [logs, wallet?.def, loadLogsPage])
 
-  useInterval(() => {
-    loadNew().catch(console.error)
-  }, 1_000, [loadNew])
+  useEffect(() => {
+    // only fetch new logs if we are on a page that uses logs
+    const needLogs = router.asPath.startsWith('/settings/wallets') || router.asPath.startsWith('/wallet/logs')
+    if (me && needLogs) {
+      const interval = setInterval(() => {
+        loadNew().catch(console.error)
+      }, 1_000)
```

Comment:
Oh, yes, I think `setTimeout` is better here for the reason you mentioned. Thanks!

Reply:
Mhh, since `loadNew` immediately changes `loadNew`, we actually cleanup the effect after every poll and run a new effect. So we need to use `timeoutId = setTimeout(reload, 1000)` in the outer scope, too.

However, because of that "effect" (pun intended), we don't need to queue a next poll ourselves anyway, see 3eba346.

I hope that is not effect abuse, lol.

Reply:
Nice solution.
Maybe needLogs should be its own state
```
useEffect(() => {
    // only fetch new logs if we are on a page that uses logs
    setNeedLogs(me && (router.asPath.startsWith('/settings/wallets') || router.asPath.startsWith('/wallet/logs')))    
}, [me?.id, router.pathname])
```

 so that the timer doesn't reset multiple times (that would be a problem if it was to happen right after it ticks) when you navigate between pages that need logs.

I think there is still a possibility for two loadNew (one of which is stale) to run at the same time, that would end up in logs being duplicated in some situations.
But to fix that you would need to abort after the event loop is yield here: 
` const result = await loadLogsPage(1, logsPerPage, wallet?.def, variables)`
but it is not worth to spaghetti the code just for a theoretical race condition that is non-critical and needs extreme bad luck to happen, or at least that's what i think. Just leaving here a note so that if for some reason we see this happening in the wild, we know why.

Reply:
> I think there is still a possibility for two loadNew (one of which is stale) to run at the same time, that would end up in logs being duplicated in some situations

I don't think this would be a problem since we always deduplicate logs. So it also wasn't a problem before.